pkg/files/manager_test.go

```diff
@@ -344,3 +345,80 @@ func TestUpdateFile_Failure(t *testing.T) {
 		})
 	}
 }
+
+func TestListFile(t *testing.T) {
+	// SET UP
+	fs := afero.NewMemMapFs()
+	for _, file := range []struct {
+		path    string
+		content string
+	}{
+		{
+			path:    "/hello.txt",
+			content: "Hi there\n",
+		},
+		{
+			path:    "/something/something.txt",
+			content: "something1\nsomething2\nsomething3\n",
+		},
+		{
+			path:    "/something/items.json",
+			content: `["a1","a2"]`,
+		},
+		{
+			path:    "/node_modules/module1/file.js",
+			content: `import tmp`,
+		},
+		{
+			path:    "/node_modules/module2/file.js",
+			content: `import tmp`,
+		},
+	} {
+		if err := afero.WriteFile(fs, file.path, []byte(file.content), 0o644); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// RUN test
+	for _, tt := range []struct {
+		name       string
+		filter     files.PatternFilter
+		showHidden bool
+		wantPath   []string
+	}{
+		{
+			name:       "all files",
+			filter:     files.PatternFilter{},
+			showHidden: false,
+			wantPath:   []string{"/hello.txt", "/something/something.txt", "/something/items.json", "/node_modules/module1/file.js", "/node_modules/module2/file.js"},
+		},
+		{
+			name: "with exclude filter",
+			filter: files.PatternFilter{
+				Include: []string{"*something*"},
+				Exclude: []string{"*.json", "node_modules"},
+			},
+			showHidden: false,
+			wantPath:   []string{"/something/something.txt"},
+		},
```

Comment:
I tried adding this test case but it failed:
```
{
	name: "include directory",
	filter: files.PatternFilter{
		Include: []string{"something"},
	},
	showHidden: false,
	wantPath:   []string{"/something/something.txt", "/something/items.json"},
},
```

Reply:
Making a pattern more specific doesn't help `/something/*`

Reply:
I will sort this out today. As discussed IRL the issue is that we use base path for matching. It can be convenient however it's not flexible enough. F.ex. `/root/something/folder/something/file1.txt` we can not ignore second `something` by providing `/root/something/something/*` also probably will be confusing for the users.